virtwho/register.py

```diff
@@ -1,5 +1,281 @@
+            return output
+        else:
+            raise FailException(f'Failed to register {self.host}')
+
+    def unregister(self):
+        """
+        Unregister and clean host by subscription-manager.
+        """
+        ret, _ = self.ssh.runcmd('subscription-manager unregister;'
+                                 'subscription-manager clean')
+        if ret == 0:
+            logger.info(f'Succeeded to unregister host')
+        else:
+            raise FailException(f'Failed to unregister {self.host}')
+
+    def satellite_cert_install(self):
+        """
+        Install certificate when registering to satellite.
+        """
+        if 'satellite' in self.register_type:
+            cmd = f'rpm -ihv http://{self.server}' \
+                  f'/pub/katello-ca-consumer-latest.noarch.rpm'
+            ret, _ = self.ssh.runcmd(cmd)
+            if ret != 0:
+                raise FailException(
+                    f'Failed to install satellite certification for {self.host}')
+
+    def attach(self, pool=None, quantity=None):
+        """
+        Attach subscription by Pool ID or --auto.
+        :param pool: Pool ID, attach by --auto when pool=None
+        :param quantity: subscription number to attach, default is auto.
+        :return: tty output.
+        """
+        cmd = 'subscription-manager attach '
+        if pool:
+            cmd += f'--pool={pool} '
+        if quantity:
+            cmd += f'--quantity={quantity}'
+        if not pool:
+            cmd += f'--auto '
+        self.refresh()
+        ret, output = self.ssh.runcmd(cmd)
+        if ret == 0:
+            logger.info(f'Succeeded to attach subscription for {self.host}')
+            return output.strip()
+        if '--auto' in cmd and 'Unable to find available' in output:
+            logger.warning(
+                f'Failed to attach subscription by auto for {self.host}.')
+            return output.strip()
+        if 'Multi-entitlement not supported' in output:
+            logger.warning(output)
+            return output.strip()
+        else:
+            raise FailException(f'Failed to attach subscription for {self.host}')
+
+    def unattach(self, pool=None):
+        """
+        Remove subscription by Pool ID or remove all.
+        :param pool: Pool ID, remove all when pool=None.
+        """
+        cmd = 'subscription-manager remove --all'
+        if pool:
+            cmd = f'subscription-manager remove --pool={pool}'
+        ret, output = self.ssh.runcmd(cmd)
+        if ret == 0:
+            logger.info(f'Succeeded to remove subscription for {self.host}')
+        else:
+            raise FailException(f'Failed to remove subscription for {self.host}')
+
+    def available(self, sku_id, virtual=False):
+        """
+        Search and analyze an available subscription by name and type.
+        :param sku_id: sku id, such as RH00001
+        :param virtual: sku type, 'Physical' or 'Virtual'.
```

Comment:
how about to change the option 'virtual' to 'sku_type'?
the tester should define the sku_type to 'virtual' or 'physical' self.

Reply:
It was the design at beginning and about why I changed to use the ```virtual=True/False``` is because the user must make sure inputting the exact ```Virtual``` or ```Physical```. 
But I also changed it as your suggestion because it's really more readable.